core/catalog-core-api/src/main/java/ddf/catalog/CatalogFrameworkImpl.java

```diff
@@ -1044,13 +1044,13 @@ private QueryResponse doQuery(QueryRequest queryRequest, FederationStrategy stra
      */
     protected QueryResponse addProcessingDetails(Set<ProcessingDetails> exceptions,
             QueryResponse response) {
-        Set<ProcessingDetails> sourceDetails = response.getProcessingDetails();
         if (!exceptions.isEmpty()) {
             // we have exceptions to merge in
-            if (sourceDetails == null) {
+            if (response == null) {
                 logger.error("Could not add Query exceptions to a QueryResponse because the list of ProcessingDetails was null -- according to the API this should not happen");
```

Comment:
I also put the check for null on sourceDetails back in as well.  So now it checks both response and sourceDetails.